element_interface/intanloader/rhsutilities.py

```diff
@@ -0,0 +1,1037 @@
+        ] = np.array(struct.unpack("<" + "H" * 128, fid.read(256)))
+
+
+# Define data_to_result function
+def data_to_result(header, data, data_present):
+    """Moves the header and data (if present) into a common object."""
+
+    result = {}
+
+    stim_parameters = {}
+    stim_parameters["stim_step_size"] = header["stim_step_size"]
+    stim_parameters["charge_recovery_current_limit"] = header["recovery_current_limit"]
+    stim_parameters["charge_recovery_target_voltage"] = header[
+        "recovery_target_voltage"
+    ]
+    stim_parameters["amp_settle_mode"] = header["amp_settle_mode"]
+    stim_parameters["charge_recovery_mode"] = header["charge_recovery_mode"]
+    result["stim_parameters"] = stim_parameters
+
+    result["stim_channels"] = header["stim_channels"]
+    result["spike_triggers"] = header["spike_triggers"]
+    result["notes"] = header["notes"]
+    result["frequency_parameters"] = header["frequency_parameters"]
+
+    result["reference_channel"] = header["reference_channel"]
+    result["amplifier_channels"] = header["amplifier_channels"]
+    result["board_adc_channels"] = header["board_adc_channels"]
+    result["board_dac_channels"] = header["board_dac_channels"]
+
+    result["dc_amplifier_data_saved"] = header["dc_amplifier_data_saved"]
+    if header["dc_amplifier_data_saved"]:
+        result["dc_amplifier_channels"] = header["dc_amplifier_channels"]
+
+    result["compliance_limit_channels"] = header["compliance_limit_channels"]
+    result["charge_recovery_channels"] = header["charge_recovery_channels"]
+    result["amp_settle_channels"] = header["amp_settle_channels"]
+    result["board_dig_in_channels"] = header["board_dig_in_channels"]
+    result["board_dig_out_channels"] = header["board_dig_out_channels"]
+
+    if data_present > 0:
+        result["t"] = data["t"]
+        result["stim_data"] = data["stim_data"]
+        result["amplifier_data"] = data["amplifier_data"]
+        result["board_adc_data"] = data["board_adc_data"]
+        result["board_dac_data"] = data["board_dac_data"]
+        if header["dc_amplifier_data_saved"] > 0:
+            result["dc_amplifier_data"] = data["dc_amplifier_data"]
+        result["compliance_limit_data"] = data["compliance_limit_data"]
+        result["charge_recovery_data"] = data["charge_recovery_data"]
+        result["amp_settle_data"] = data["amp_settle_data"]
+        result["board_dig_in_data"] = data["board_dig_in_data"]
+        result["board_dig_out_data"] = data["board_dig_out_data"]
+
+    return result
+
+
+# Define plot_channel function
+def plot_channel(channel_name, result):
+    # Find channel that corresponds to this name
+    channel_found, signal_type, signal_index = find_channel_in_header(
+        channel_name, result
+    )
+
+    # Plot this channel
+    if channel_found:
+        fig, ax = plt.subplots()
+        ax.set_title(channel_name)
+        ax.set_xlabel("Time (s)")
+
+        if signal_type == "amplifier_channels":
```

Comment:
This looks to me like it could be a dict

```python
signal_type_lookup = dict(
  amplifier_channels = ["Voltage", "amplifier_data"]
  "dc_amplifier_channels" = ...
)
ylabel, signal_data_name = signal_type_lookup.get(signal_type, [None, None])
if not ylabel:
  raise ValueError("Err message")
```